rpc2/codec.go

```diff
@@ -1,5 +1,55 @@
 package rpc2
 
+import (
+	"bytes"
+	"github.com/ugorji/go/codec"
+	"io/ioutil"
+)
+
 type Decoder interface {
 	Decode(interface{}) error
 }
+
+type Encoder interface {
+	Encode(interface{}) error
+}
+
+type ByteEncoder interface {
+	EncodeToBytes(interface{}) ([]byte, error)
+}
+
+type FramedMsgpackEncoder struct {
+	handle codec.Handle
+}
+
+func NewFramedMsgpackEncoder(mh *codec.MsgpackHandle) *FramedMsgpackEncoder {
+	if mh == nil {
```

Comment:
I've changed it so the handle isn't passed around at all, instead each owner creates one itself. I think that better encapsulates functionality that should be hidden away.